commons-core/src/main/scala/com/avsystem/commons/serialization/cbor/CborAdtMetadata.scala

```diff
@@ -253,3 +253,17 @@ trait CborAdtInstances[T] {
   def cborCodec: GenObjectCodec[T] =
     metadata.setup(_.validate()).adjustCodec(stdCodec)
 }
+
+trait CborAdtPolyInstances[C[_]] {
+  def stdCodec[T: GenCodec]: GenObjectCodec[C[T]]
+  def metadata[T: GenCodec]: CborAdtMetadata[C[T]]
+  def cborCodec[T: GenCodec]: GenObjectCodec[C[T]] =
+    metadata.setup(_.validate()).adjustCodec(stdCodec)
```

Comment:
We can't do validation every single time the codec is requested. Note: this was not a problem with the non-poly version of this class because the codec is saved into a `val` but here we have a `def`.

Reply:
fixed